### PR TITLE
Improvements for Badge component

### PR DESCRIPTION
**Pull request by EduardTrutsyk**
#### What does this PR do:
* added new `prop`
* fixed bug
* updated documentation

b48088d83c6f95cddb8bff83285d59c0

### CODE REVIEW

File: components/badge/badge.md [hunk 1]
```diff
@@ -1,57 +1,35 @@
...
+      isVisibleTop: false,
+      isVisibleRight: true,
+      isVisibleBottom: true,
+      isVisibleLeft: true,
+      isArrow: false,
+      isBorder: true,
+    };
+
+    <div>
+      <Checkbox name="isVisibleTop" checked={state.isVisibleTop} onChange={() => setState({ isVisibleTop: !state.isVisibleTop })}>top</Checkbox>
```
Comment (yurist38) [inline components/badge/badge.md]: What if reformat all those single lines components to multi lines?
```
<Checkbox
  name="isVisibleTop"
  checked={state.isVisibleTop}
  onChange={() => setState({ isVisibleTop: !state.isVisibleTop })}
>
  top
</Checkbox>
```
It's gonna be much easier to read and maintain I think...

File: tests/unit/badge/__snapshots__/badge.spec.js.snap [hunk 2]
```diff
@@ -1,5 +1,14 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`Badge should render content without badge when badge is not visible 1`] = `
+<div
```
Comment (yurist38) [inline tests/unit/badge/__snapshots__/badge.spec.js.snap]: But here is the opposite thing: when we have just one attribute I don't think that we should place it to separate line.
`<div className="ui-badge">` looks good enough.

Comment (EduardTrutsyk) [inline tests/unit/badge/__snapshots__/badge.spec.js.snap]: but it's snapshot, which generated automatically, I can't change it ;)

Comment (yurist38) [inline tests/unit/badge/__snapshots__/badge.spec.js.snap]: Ah, sorry. Didn't notice that.

File: components/badge/badge.js [hunk 3]
```diff
@@ -23,20 +23,20 @@ const Badge = ({ arrow, border, children, mods, position, title, ...otherProps }
...
     }
 
-    badge = (
+    badge = visible ? (
       <div {...otherProps} className={getClassNamesWithMods('ui-badge-badge', mods, badgeMods)}>
         {title}
         {arrow && (position === 'right' || position === 'left') ? <span className="ui-badge-badge-arrow" /> : null}
       </div>
-    );
+    ) : <noscript />;
```
Comment (asci) [inline components/badge/badge.js]: AFAIK could be just `null`

**Pull request merged**